src/resources.js

```diff
@@ -20,7 +21,12 @@ function get(...pathParts) {
  * @param {String} name The name of the binary, without file extension.
  */
 function executable(name) {
-    return get(os.platform(), /^win/i.test(os.platform()) ? `${name}.exe` : name);
+    let execPath = get(os.platform(), /^win/i.test(os.platform()) ? `${name}.exe` : name);
+    if (fs.existsSync(execPath)) {
```

Comment:
We shouldn't add more uses of sync fs functions.

Reply:
Actually, `resources.executable` is always called from within an async function or a Promise, so this is possible.

Reply:
FWIW, I just saw that node actually has built-in promise versions of the fs functions, such as [`fsPromise.access`](https://nodejs.org/api/fs.html#fs_fspromises_access_path_mode) (`fs.exists` is deprecated).

Reply:
See commit b254536

Reply:
We should keep that for future advisement. The docs say there's no reason to use `fs` directly.